### DISCUSSION TITLE
BackgroundTasks does not throw an error if you miss passing in parameters to the function, just runs forever

### DISCUSSION THREAD

**Discussion by johnathanchiu** [fastapi/fastapi#12775] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI

app = FastAPI()

async def background_task_fn(param1, param2):
    # DO SOMETHING
    pass

@app.post("/fn")
def fn(param1, param2) -> Response:
    # NOTICE HERE HOW `param1` IS MISSING
    background_tasks.add_task(background_task_fn, param2) 
    return Response(content="Function is running", media_type="text/plain")
```

### Description

The above pseudocode highlights what I am trying to achieve. The background task above will never throw an error and this is a bit annoying since I spent a lot of time debugging just to realize this. How could I get it to check properly?

### Operating System

macOS

### FastAPI Version

0.115.2

### Pydantic Version

2.9.2

### Python Version

Python 3.11.10

**Comment by YuriiMotov** [comment, marked as answer]
mypy can help you find such errors:
```
from fastapi import BackgroundTasks, FastAPI, Response

app = FastAPI()

async def background_task_fn(param1: str, param2: int):
    print("Background task is started")

@app.post("/fn")
def fn(background_tasks: BackgroundTasks):
    # NOTICE HERE HOW `param1` IS MISSING
    background_tasks.add_task(background_task_fn, 1)
    return Response(content="Function is running", media_type="text/plain")
```

Mypy output:
```
t.py:11: error: Missing positional argument "param2" in call to "add_task" of "BackgroundTasks"  [call-arg]
t.py:11: error: Argument 2 to "add_task" of "BackgroundTasks" has incompatible type "int"; expected "str"  [arg-type]
Found 2 errors in 1 file (checked 1 source file)
```


**UPD:** Actually, it's not "just runs forever". It throws the error:
```
TypeError: background_task_fn() missing 1 required positional argument: 'param2'
```

**Reply by johnathanchiu** [reply]
Good point here, I could use mypy to lint these pieces. I will add that actually to my build system.

**Comment by estebanx64** [comment]
Hi @johnathanchiu .

As @YuriiMotov said before, it won't run forever, I'm getting the same error.

`TypeError: background_task_fn() missing 1 required positional argument: 'param2'`

Can you elaborate a special case where you are seeing it running forever because of a missing param? Otherwise, if you are getting now the same error, let us know to be able to close this topic, thanks in advance

**Comment by johnathanchiu** [comment]
hmm, how long do you have to wait before it errors? I let it run for ~10 minutes yesterday and it didn't error. Since I am doing a long running task I thought maybe it was just something slow as a result of that task just to realize it was because of the parameter missing.

**Reply by YuriiMotov** [reply]
With the code example I provided above I see the error immediately

**Reply by johnathanchiu** [reply]
I see, let me look into it more. It's hard to create a minimal reproducible example in my case. I'll see what I can put together.
